drivers/fuel_gauge/ltc2959/ltc2959.c

```diff
@@ -0,0 +1,544 @@
+			return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+		}
+		case LTC2959_GPIO_MODE_UNIPOLAR: {
+			uint16_t raw = (uint16_t) (((int64_t) value_uv * 32768) /
+			        LTC2959_GPIO_UNIPOLAR_UV_SF);
+			uint8_t buf[2] = {raw >> 8, raw & 0xFF};
+			return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+		}
+		default:
+			break;
+	}
+
+	LOG_ERR("Unsupported GPIO mode: 0x%02x", gpio_mode);
+	return -ENOTSUP;
+}
+
+static int ltc2959_get_voltage_threshold_uv(const struct device* dev, bool high, uint32_t* value)
+{
+	uint8_t reg = high ? LTC2959_REG_VOLT_THRESH_HIGH_MSB : LTC2959_REG_VOLT_THRESH_LOW_MSB;
+	uint16_t raw;
+	int ret = ltc2959_read16(dev, reg, &raw);
+
+	if (ret < 0) {
+		LOG_ERR("Failed to get voltage threshold: %i", ret);
+		return ret;
+	}
+
+	*value = ((uint64_t) raw * LTC2959_VOLT_THRESH_UV_SCALAR) >> 15;
+
+	return 0;
+}
+
+static int ltc2959_set_voltage_threshold_uv(const struct device* dev, bool high, uint32_t value)
+{
+	uint8_t reg = high ? LTC2959_REG_VOLT_THRESH_HIGH_MSB : LTC2959_REG_VOLT_THRESH_LOW_MSB;
+	const uint16_t raw = (((uint64_t) value << 15) / LTC2959_VOLT_THRESH_UV_SCALAR);
+
+	uint8_t buf[2] = {raw >> 8, raw & 0xFF};
+	const struct ltc2959_config* cfg = dev->config;
+
+	return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+}
+
+static int ltc2959_get_current_threshold_ua(const struct device* dev, bool high, int32_t* value_ua)
+{
+	uint8_t reg = high ? LTC2959_REG_CURR_THRESH_HIGH_MSB : LTC2959_REG_CURR_THRESH_LOW_MSB;
+
+	uint16_t raw;
+	int ret = ltc2959_read16(dev, reg, &raw);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	const struct ltc2959_config* cfg = dev->config;
+	int16_t signed_raw = (int16_t) raw;
+	*value_ua = signed_raw * cfg->current_lsb_uv;
+
+	return 0;
+}
+
+static int ltc2959_set_current_threshold_ua(const struct device* dev, bool high, int32_t value_ua)
+{
+	uint8_t reg = high ? LTC2959_REG_CURR_THRESH_HIGH_MSB : LTC2959_REG_CURR_THRESH_LOW_MSB;
+	const struct ltc2959_config* cfg = dev->config;
+	int32_t raw32 = value_ua / cfg->current_lsb_uv;
+
+	/* To account for cases where current thresholds are +-2A */
+	int16_t raw = CLAMP(raw32, INT16_MIN, INT16_MAX);
+	uint8_t buf[2] = {raw >> 8, raw & 0xFF};
+
+	 return i2c_burst_write_dt(&cfg->i2c, reg, buf, sizeof(buf));
+}
+
+static int ltc2959_get_temp_threshold_dK(const struct device* dev, bool high, uint16_t* value_dK)
+{
+	uint8_t reg = high ? LTC2959_REG_TEMP_THRESH_HIGH_MSB : LTC2959_REG_TEMP_THRESH_LOW_MSB;
+	uint16_t raw;
+	int ret = ltc2959_read16(dev, reg, &raw);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	*value_dK = ((uint32_t) raw * LTC2959_TEMP_K_SF) >> 16;
+
+	return 0;
+}
+
+static int ltc2959_set_temp_threshold_dK(const struct device* dev, bool high, uint16_t value_dK)
+{
+	uint8_t reg = high ? LTC2959_REG_TEMP_THRESH_HIGH_MSB : LTC2959_REG_TEMP_THRESH_LOW_MSB;
+	uint16_t raw = ((uint32_t) value_dK << 16) / LTC2959_TEMP_K_SF;
+	uint8_t buf[2] = {raw >> 8, raw & 0xFF};
```

Comment:
Better use `sys_put_be16()` or from `byteorder.h`
```suggestion
	uint8_t buf[2];
	sys_put_be16(raw, buf);
```